### DISCUSSION TITLE
How to dev a workflow without making many commits to main branch?

### DISCUSSION THREAD

**Discussion by CybCom** [community/community#83572] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm currently writing CI workflows for a course assignment, and there is a problem that I have to commit changes to the main branch each time to actually run and test it, which works but also makes the history of the main branch messy. So is there any method to debug and run a GitHub Action workflow out of the main branch?

Thanks and happy new year for everyone :-)

**Comment by faridnec** [comment, marked as answer]
You can perhaps create a new branch specifically for your CI workflow development and testing. This way, you keep your main branch clean and only merge changes when you're sure they're ready.

Create a new branch from the main branch for your CI workflow changes -> Push your feature branch to the repository, and the workflow will run on this branch -> Examine the workflow run logs if there are any issues -> If everything looks good, create a pull request to merge your changes into the main branch.

I hope this helps

**Reply by CybCom** [reply]
Thanks and I just noticed the switch that decides which branch the ci file lives on😀 have a good new year!
